builtin/providers/oneandone/resource_oneandone_firewall_policy.go

```diff
@@ -0,0 +1,299 @@
+package oneandone
+
+import (
+	"fmt"
+	"github.com/1and1/oneandone-cloudserver-sdk-go"
+	"github.com/hashicorp/terraform/helper/schema"
+	"strings"
+)
+
+func resourceOneandOneFirewallPolicy() *schema.Resource {
+	return &schema.Resource{
+
+		Create: resourceOneandOneFirewallCreate,
+		Read:   resourceOneandOneFirewallRead,
+		Update: resourceOneandOneFirewallUpdate,
+		Delete: resourceOneandOneFirewallDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"rules": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"protocol": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"port_from": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
There is a new func that you can use as part of the helpers package:

```
validation.IntBetween(1, 65535),
```

it lives in the `github.com/hashicorp/terraform/helper/validation` package